spec/requests/api/v0/user_requests/update_spec.rb

```diff
@@ -0,0 +1,45 @@
+require "rails_helper"
+
+RSpec.describe "User Update Profile Endpoint" do
+  describe "happy path" do
+    it "supports with POST request for User Update", :vcr do
+      test_data
+
+      user1 = @user1
+
+      expect(user1.username).to eq("username1")
+
+      patch "/api/v0/users/#{user1.id}?username=username0" #, params: { username: "username0" }, headers: {"CONTENT_TYPE" => "application/json", "ACCEPT" => "application/json"}
+
+      expect(response).to be_successful
+      expect(response.status).to eq(200)
+
+      response_data = JSON.parse(response.body, symbolize_names: true)[:data]
+
+      expect(response_data[:type]).to eq("user")
+      expect(response_data[:id]).to be_a(String)
+
+      expect(response_data[:attributes][:email]).to eq("user1@gmail.com")
+      expect(response_data[:attributes][:username]).to eq("username0")
+    end
+  end
+
+  describe "sad path" do
+    it "returns an error if user does not exist", :vcr do
```

Comment:
I would potentially consider a couple of other SAD paths.
1. If the user params for username is left blank and the person tries to update, passing a nil/blank value.
2. If the user tries to update with a username that already exists. Again a little rusty, the model has validation for `username` as `  validates :username, presence: true, `uniqueness: {case_sensitive: false}`, so I think it has to be unique w/o taking into account case sensitivity username updating to USERNAME, would fail.

Reply:
@Sykogst I was able to get these added!